ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/capella/block/BlockProcessorCapella.java

```diff
@@ -192,34 +193,57 @@ public void processWithdrawals(
             .getWithdrawalsSchemaRequired()
             .createFromElements(getExpectedWithdrawals(state));
 
-    if (payloadSummary.getOptionalWithdrawalsRoot().isEmpty()
-        || !expectedWithdrawals
-            .hashTreeRoot()
-            .equals(payloadSummary.getOptionalWithdrawalsRoot().get())) {
-      throw new BlockProcessingException(
-          "Expected "
-              + expectedWithdrawals.hashTreeRoot()
-              + " withdrawals root, but withdrawals root was "
-              + (payloadSummary.getOptionalWithdrawalsRoot().isPresent()
-                  ? payloadSummary.getOptionalWithdrawalsRoot().get()
-                  : "MISSING"));
-    }
+    assertWithdrawalsInExecutionPayloadMatchExpected(payloadSummary, expectedWithdrawals);
+
     for (int i = 0; i < expectedWithdrawals.size(); i++) {
       final Withdrawal withdrawal = expectedWithdrawals.get(i);
       beaconStateMutators.decreaseBalance(
           state, withdrawal.getValidatorIndex().intValue(), withdrawal.getAmount());
     }
 
-    if (expectedWithdrawals.size() > 0) {
+    if (expectedWithdrawals.size() != 0) {
```

Comment:
this is where we'll need to be careful... 0 is definitely possible in capella...

I'll have to recheck the spec change, but my expectation is that 
 - if size == max, next = last_found+1
 - otherwise next increments prestate.next + maxValidatorsPerSweep

Reply:
You were right here - this if structure didn't match the spec correctly and we wound up not calling `setNextWithdrawalValidatorIndex` on the state if there were no withdrawals.  The condition was right, but it should have only been around the `setNextWithdrawalIndex` part not the whole rest of the method.

Will push up a fix in a bit.

Reply:
Nice catch Paul!!